pkg/providers/pricing/pricing.go

```diff
@@ -38,13 +38,13 @@ import (
 	"github.com/aws/karpenter-core/pkg/utils/pretty"
 )
 
-// PricingProvider provides actual pricing data to the AWS cloud provider to allow it to make more informed decisions
+// Provider provides actual pricing data to the AWS cloud provider to allow it to make more informed decisions
```

Comment:
We should think about doing a similar model here to our cluster state where we have a internal data representation and then we have a controller that is informing that data representation. 

This code was added before we had the `controller.SingletonController`. Now that we have that, I think that we can leverage that here. Probably a good follow-up PR.